Src/structs.c

```diff
@@ -54,7 +56,17 @@ putUname(FILE *fd, UType *tmp)
 
 	if (!tmp) return;
 	putUname(fd, tmp->nxt); /* postorder */
-	fprintf(fd, "struct %s { /* user defined type */\n",
+	char* type_string;
+	switch (tmp->type) {
+		case STRUCT:
+			type_string = "struct";
+		break;
+		case UNION:
+			type_string = "union";
+		break;
+	}
+	fprintf(fd, "%s %s { /* user defined type */\n",
+		type_string,
```

Comment:
type_string might be unassigned here. `default:` ?

or maybe:
`const char* const type_string = (tmp->type == UNION) ? "union" : "struct";` (assumes correct arguments)

Reply:
I prefer to go with default

Reply:
with previous assert this one is no longer necessary

Reply:
So what do you propose?

Reply:
as you wish, you can remove it, you can leave it as additional doc.